packages/infra/infra-core/src/lib/patterns/ecr-sync/ecr-sync.ts

```diff
@@ -0,0 +1,207 @@
+
+    const artifactsBucket = new s3.Bucket(this, 'ArtifactBucket', {
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      enforceSSL: true,
+      versioned: true,
+    });
+
+    const buildRole = new iam.Role(this, 'buildRole', {
+      assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
+    });
+
+    const lambaFile = `${path.resolve(__dirname)}/lambda/get-image-tags-handler`;
+    const entry = lambaFile + (fs.existsSync(`${lambaFile}.ts`) ? '.ts' : '.js');
+
+    const lambda = new lnjs.NodejsFunction(this, 'lambda', {
+      functionName: props.getImageTagsFunctionName,
+      entry: entry,
+      runtime: lmbd.Runtime.NODEJS_18_X,
+      timeout: cdk.Duration.minutes(10),
+      logRetention: logs.RetentionDays.ONE_WEEK,
+      memorySize: 256,
+      environment: {
+        AWS_ACCOUNT_ID: cdk.Stack.of(this).account,
+        REGION: cdk.Stack.of(this).region,
+        IMAGES: JSON.stringify(props.dockerImages),
+        REPO_PREFIX: props.repoPrefix ?? '',
+        BUCKET_NAME: artifactsBucket.bucketName,
+      },
+    });
+    artifactsBucket.grantPut(lambda);
+
+    new evt.Rule(this, 'ScheduleGetImageTags', {
+      targets: [new trgt.LambdaFunction(lambda)],
+      schedule: props.schedule ?? evt.Schedule.rate(cdk.Duration.days(1)),
+    });
+
+    props.dockerImages.forEach(element => {
+      const repoName = (props.repoPrefix) ? `${props.repoPrefix}/${element.imageName}` : element.imageName;
+      const repo = new ecr.Repository(this, element.imageName, {
+        repositoryName: repoName,
+      });
+      repo.grantPullPush(buildRole);
+
+      lambda.addToRolePolicy(new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['ecr:ListImages'],
+        resources: [repo.repositoryArn],
+      }));
+
+      if (props.lifcecyleRule !== undefined) {
+        repo.addLifecycleRule(props.lifcecyleRule);
+      }
+
+      this.ecrRepos.push(repo);
+    });
+
+    // Get triggered by s3
+    const triggerStageArtifact = new cp.Artifact();
+    const triggerAction = new cpa.S3SourceAction({
+      actionName: 'S3Source',
+      bucket: artifactsBucket,
+      bucketKey: 'images.zip',
+      output: triggerStageArtifact,
+    });
+
+    const buildSpecBuild = new cb.PipelineProject(this, 'buildSpecBuild', {
+      buildSpec: cb.BuildSpec.fromObject({
+        version: '0.2',
+        phases: {
+          install: {
+            'runtime-versions': {
+              docker: 18,
+            },
+            'commands': [
+              'aws --version',
+              'curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"',
+              'unzip -q awscliv2.zip',
+              './aws/install',
+              'aws --version',
+            ],
+          },
+          build: {
+            commands: [
+              ` set -e\n \
+                ${props.initScript || ''}\n \
+                cat images.csv\n \
```

Comment:
Same here, I'd also make it a constant bc it's also referenced inside `packages/infra/infra-core/src/lib/patterns/ecr-sync/lambda/get-image-tags-handler.ts`